src/components/articles/article-detail.js

```diff
@@ -0,0 +1,76 @@
+import PropTypes from "prop-types";
+import React, { useContext, useEffect } from "react";
+import { connect } from "react-redux";
+import { Link } from "react-router-dom";
+import { bindActionCreators } from "redux";
+import AuthContext from "../../context/auth-context";
+import { deleteArticle, loadArticles } from "../../redux/actions/article-actions";
+import { loadUsers } from "../../redux/actions/user-actions";
+
+function ArticleDetail({ loadUsers, loadArticles, deleteArticle, users, articles, article, history }) {
+  const { authenticated } = useContext(AuthContext);
+
+  useEffect(() => {
+    if (users.length === 0) loadUsers().catch((error) => alert(error));
+    if (articles.length === 0) loadArticles().catch((error) => alert(error));
+  }, []);
+
+  function onDelete() {
+    history.push("");
+    deleteArticle(article.id).catch((error) => alert(error));
+  }
+
+  function onEdit() {
+    const path = "/articles/" + article.id + "/edit";
+    history.push(path);
+  }
+
+  return (
+    <div>
+      <h2>{article.title}</h2>
+      {<p>{article.content}</p>}
+      <div>
+        <button onClick={onEdit} disabled={!authenticated}>
```

Comment:
Just disabling the edit button isn't enough to stop unsigned users because the page is still accessible to anyone who enters the url manually. We need to make the route itself only accessible to signed users.